st2common/st2common/triggers.py

```diff
@@ -32,75 +33,83 @@
-
-
-def _get_trigger_type_url(triggertype_ref):
-    base_url = get_url_without_trailing_slash(TRIGGER_TYPE_ENDPOINT)
-    return '%s/%s' % (base_url, triggertype_ref)
-
-
-def _do_register_internal_trigger_types():
-    LOG.debug('Registering internal trigger types...')
-
-    for resource_name, trigger_definitions in INTERNAL_TRIGGER_TYPES.items():
-        for trigger_definition in trigger_definitions:
-            LOG.debug('Registering internal trigger: %s', trigger_definition['name'])
-            register_trigger_type(trigger_definition=trigger_definition, attempt_no=0)
-
-
-def _is_triggertype_exists(ref):
-    try:
-        r = requests.get(url=_get_trigger_type_url(ref))
-        if r.status_code == httplib.OK:
-            return True
-    except:
-        return False
-
-
-def register_trigger_type(trigger_definition, attempt_no=0):
-    LOG.debug('Attempt no %s to register trigger %s.', attempt_no, trigger_definition['name'])
-
-    ref = ResourceReference.to_string_reference(pack=trigger_definition['pack'],
-                                                name=trigger_definition['name'])
-    if _is_triggertype_exists(ref):
-        return
-
-    payload = json.dumps(trigger_definition)
-
-    try:
-        r = requests.post(url=TRIGGER_TYPE_ENDPOINT, data=payload,
-                          headers=HTTP_POST_HEADER, timeout=TIMEOUT)
-        if r.status_code == httplib.CREATED:
-            LOG.info('Registered trigger %s.', trigger_definition['name'])
-        elif r.status_code == httplib.CONFLICT:
-            LOG.info('Trigger %s is already registered.', trigger_definition['name'])
-        else:
-            LOG.error('Seeing status code %s on an attempt to register trigger %s.',
-                      r.status_code, trigger_definition['name'])
-    except requests.exceptions.ConnectionError:
-        if attempt_no < MAX_ATTEMPTS:
-            retry_wait = RETRY_WAIT * (attempt_no + 1)
-            LOG.debug('    ConnectionError. Will retry in %ss.', retry_wait)
-            eventlet.spawn_after(retry_wait, register_trigger_type,
-                                 trigger_definition=trigger_definition,
-                                 attempt_no=(attempt_no + 1))
-        else:
-            LOG.warn('Failed to register trigger %s. Exceeded max attempts to register trigger.',
-                     trigger_definition['name'])
-    except:
-        LOG.exception('Failed to register trigger %s.', trigger_definition['name'])
 
+class InternalTriggerTypesRegistrar(object):
+
+    def __init__(self):
+        self._action_sensor_enabled = cfg.CONF.action_sensor.enable
+        self._trigger_type_endpoint = cfg.CONF.action_sensor.triggers_base_url
+        self._retry_wait = cfg.CONF.action_sensor.retry_wait
+        self._timeout = cfg.CONF.action_sensor.request_timeout
+        self._max_attempts = cfg.CONF.action_sensor.max_attempts
+        self._temporary_token = create_token('system.internal_trigger_registrar',
```

Comment:
Yeah. Set it to one hour in my recent commit. We anyways delete the token at the end. 5 mins might be too small IMO. We'll have catch unauthorized code and renegotiate a token. Seems too much code :P.